core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -127,6 +148,13 @@ SdkPojo unmarshall(XmlUnmarshallerContext context, SdkPojo sdkPojo, XmlElement r
         return (SdkPojo) ((Buildable) sdkPojo).build();
     }
 
+    private boolean isXmlString(String payload) {
+        String pattern = "^<([a-zA-Z_][\\w\\-]*)>(.*)</\\1>$";
+        Pattern xmlPattern = Pattern.compile(pattern);
```

Comment:
Pattern.compile is super-slow. We'd definitely want to do that in a static field.

Reply:
+1, pending decision on whether to parse XML formatted Strings